src/tiled/objectsdock.h

```diff
@@ -57,23 +58,32 @@ private slots:
     void documentAboutToClose(Document *document);
     void moveObjectsUp();
     void moveObjectsDown();
+    void filterObjects();
 
 private:
     void retranslateUi();
 
-    void saveExpandedGroups();
-    void restoreExpandedGroups();
+    void saveExpandedGroups(QMap<MapDocument*, QList<ObjectGroup*> > &expansionState);
+    void restoreExpandedGroups(QMap<MapDocument*, QList<ObjectGroup*> > &expansionState);
 
     QAction *mActionNewLayer;
     QAction *mActionObjectProperties;
     QAction *mActionMoveToGroup;
     QAction *mActionMoveUp;
     QAction *mActionMoveDown;
 
+    QLineEdit *mFilterEdit;
     ObjectsView *mObjectsView;
     MapDocument *mMapDocument;
+
+    // These maps store the state of the objects dock for each opened map document
     QMap<MapDocument*, QList<ObjectGroup*> > mExpandedGroups;
+    QMap<MapDocument*, QList<ObjectGroup*> > mPrefilterExpandedGroups;
+    QMap<MapDocument*, QString> mFilterStrings;
+    QMap<MapDocument*, bool> mFilterWasEmpty;
+
     QMenu *mMoveToMenu;
+    QModelIndex getGroupIndex(ObjectGroup *og);
```

Comment:
Would be called `groupIndex` according to naming style, but I think it should instead be a convenience override to the previously suggested `mapToViewModel` helper function. Should be marked `const`.